Fix unused-parameter warnings, unnecessary unused attributes

norihiro opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst -->

### Description
<!--- Describe your changes in detail. -->
<!--- If this change includes UI elements, please include screenshots. -->
- Add `OBS_UNUSED` attribute to unused function parameters.
- Remove `UNUSED_PARAMETER` from used parameters
- Remove `OBS_UNUSED` from a used parameter

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open GitHub Issue, or implements feature request -->
<!--- from the Ideas page, please link to the issue here. -->
Don't want to see warnings.
Don't want to see unnecessary `UNUSED_PARAMETER`.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment (hardware, OS version, etc.),-->
<!--- and the tests you ran, including how it may affect other areas of code. -->
- Reviewed code to ensure the changes are just touching `OBS_UNUSED` and `UNUSED_PARAMETER`.
- Checked warnings built on Linux, macOS, and Windows.

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
<!--- - Bug fix (non-breaking change which fixes an issue) -->
<!--- - New feature (non-breaking change which adds functionality) -->
<!--- - Tweak (non-breaking change to improve existing functionality) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
- Code cleanup (non-breaking change which makes code smaller or more readable)
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
<!--- - Documentation (a change to documentation pages) -->

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

tytan652 commented on plugins/obs-filters/noise-suppress-filter.c:
```diff
@@ -340,8 +340,6 @@ static inline void alloc_channel(struct noise_suppress_data *ng,
 #ifdef LIBSPEEXDSP_ENABLED
 	ng->spx_states[channel] =
 		speex_preprocess_state_init((int)frames, sample_rate);
-#else
-	UNUSED_PARAMETER(sample_rate);
```
The related function definition should not have `OBS_UNUSED` since it is used if speexdsp is enabled. So it would be better to remove the `OBS_UNUSED` and keep the `UNUSED_PARAMETER.`

tytan652 replied:
Don't forget to mark this "review" as resolved.

jp9000 commented:
You don't need to mark unused parameters in function definitions in C++. You can just remove the parameter's name. (Can't do that in C though, yet)

RytoEX reviewed (changes requested):
The change to remove variables in the first commit I would expect to be on its own.

The fourth commit has a change/revert to obs-filters which should be applied to the third commit to revert the change there.

Otherwise, seems fine.

norihiro commented:
I have updated as Jim and RytoEX commented.
- Unused parameters are removed from `OBSBasic::LoadTransitions`.
- Separate the commit to remove `ms = ticks * AUDIO_OUTPUT_FRAMES * 1000 / sample_rate;` into standalone commit.
- The commits to `plugins/obs-filters/noise-suppress-filter.c` are reordered so that there is no change-then-revert in the series of commits.

RytoEX commented on UI/window-basic-main-transitions.cpp:
```diff
@@ -1770,8 +1770,7 @@ obs_data_array_t *OBSBasic::SaveTransitions()
 	return transitions;
 }
 
-void OBSBasic::LoadTransitions(obs_data_array_t *transitions,
-			       obs_load_source_cb cb, void *private_data)
```
I feel like this change is out of scope for this PR, as we're changing a function signature, even though it's private. I'd be fine with this and its related changes as its own PR.

RytoEX reviewed (changes requested):
I think I'd prefer the function signature change with `OBSBasic::LoadTransition` to be its own PR.

norihiro commented:
After rebasing, I saw some more unused parameter warnings. I will update this PR.

norihiro commented:
Added these changes after Ryan's review comment.
- File `libobs/util/bmem.c` function `base_set_allocator`: Added `OBS_UNUSED`.
- File `plugins/linux-capture/xcomposite-input.c` function `silence_x11_errors`: Added `OBS_UNUSED`.
- File `plugins/vlc-video/vlc-video-source.c` function `init_screencast_capture`: Removed a local variable `aux`, which is not set, not read at all.
- File `plugins/vlc-video/vlc-video-source.c` function `vlcs_video_format`: Removed two local variables `new_width` and `new_height`, which are initialized to 0 but not set, not read.

RytoEX reviewed (approved):
CI built on all jobs. Visually, this seems fine to me. This is probably fine to merge. Will circle back around to this to merge if no one raises any concerns.

tytan652 commented:
On macOS (or with Clang), `LoadTransitions` emits unused-parameter warning.

```diff
diff --git a/UI/window-basic-main-transitions.cpp b/UI/window-basic-main-transitions.cpp
index 629506eb557d..de2492f68bad 100644
--- a/UI/window-basic-main-transitions.cpp
+++ b/UI/window-basic-main-transitions.cpp
@@ -1771,7 +1771,8 @@ obs_data_array_t *OBSBasic::SaveTransitions()
 }
 
 void OBSBasic::LoadTransitions(obs_data_array_t *transitions,
-			       obs_load_source_cb cb, void *private_data)
+			       OBS_UNUSED obs_load_source_cb cb,
+			       OBS_UNUSED void *private_data)
 {
 	size_t count = obs_data_array_count(transitions);
 

```

norihiro commented:
> On macOS (or with Clang), `LoadTransitions` seems to emit unused-parameter warning.

Thank you for reviewing this PR. The other PR (#5148) was just now merged and it fixed `LoadTransitions`.
